Enable llama2-70b LoRA finetuning (#527)
huggingface/optimum-habana: examples/language-modeling/README.md

## PEFT

> To run LoRA finetuning, you can use `run_lora_clm.py`. Multi-card examples can be simply adapted to run LoRA finetuning.
> Here are command examples for Llama1-7B, Falcon-40B and Llama2-70B:

Reviewer: ```suggestion
To run LoRA finetuning, you can use `run_lora_clm.py`.
Here are single-/multi-device command examples for Llama1-7B, Falcon-40B and Llama2-70B:
```
Let's change that too, it sounds a bit redundant.

Author: Agree. addressed

    --low_cpu_mem_usage True
```

- Multi-card finetuning of Llama2-70B with DeepSpeed ZeRO-3 optimization and LoRA:

>   The following command requires Habana DeepSpeed 1.13.0 or later.

Reviewer: ```suggestion
  > The following command requires Habana DeepSpeed 1.13.0 or later.
```

    --low_cpu_mem_usage True
```

- Multi-card finetuning of Llama2-70B with DeepSpeed ZeRO-3 optimization and LoRA:

>   **Note:** the following command requires Habana DeepSpeed 1.13.0 or later.

Reviewer: ```suggestion
  > The following command requires Habana DeepSpeed 1.13.0 or later.
```

```bash
PT_HPU_MAX_COMPOUND_OP_SIZE=10 DEEPSPEED_HPU_ZERO3_SYNC_MARK_STEP_REQUIRED=1 \
python3 ../gaudi_spawn.py --use_deepspeed  --world_size 8  run_lora_clm.py \

>   --model_name_or_path path_to_llama2_70b_ckpt/Llama-2-70b-hf \

Reviewer: ```suggestion
  --model_name_or_path meta-llama/Llama-2-70b-hf \
```

  --deepspeed path_to_zero3_config.json \
  --dataset_name tatsu-lab/alpaca \
  --bf16 True \
  --output_dir ./lora_out \
  --num_train_epochs 2 \
  --max_seq_len 2048 \
  --per_device_train_batch_size 10 \
  --per_device_eval_batch_size 10 \
  --gradient_checkpointing \
  --evaluation_strategy epoch \
  --eval_delay 2 \
  --save_strategy no \
  --learning_rate 0.0018 \
  --warmup_ratio 0.03 \
  --lr_scheduler_type "cosine" \
  --logging_steps 1 \
  --dataset_concatenation \
  --attn_softmax_bf16 True \
  --do_train \
  --do_eval \
  --use_habana \
  --use_lazy_mode \
  --pipelining_fwd_bwd \
  --throughput_warmup_steps 3 \
  --lora_rank 4 \

>   --lora_target_modules "q_proj" "v_proj" 'k_proj' 'o_proj' \

Reviewer 2: just a minor thing. q, v are in double quotes and the rest use single quotes

Author: Good point. fixed

  --validation_split_percentage 4
````

Refer to this [documentation](https://huggingface.co/docs/optimum/habana/usage_guides/deepspeed) for more information about how to use DeepSpeed in Optimum Habana.
Here is a DeepSpeed ZeRO-3 configuration you can use to fine-tune Llama2-70B on 8x Gaudi2 cards:

```json
{
    "steps_per_print": 64,
    "train_batch_size": "auto",
    "train_micro_batch_size_per_gpu": "auto",
    "gradient_accumulation_steps": "auto",
    "bf16": {
        "enabled": true
    },
    "gradient_clipping": 1.0,
    "zero_optimization": {
        "stage": 3,
        "overlap_comm": false,
        "contiguous_gradients": false
    }
}

> ```

Author: Agree. Changed
